htdocs/public/payment/newpayment.php

```diff
@@ -1689,10 +1759,7 @@ class="stripe-button"
 	    }
 	    </style>';
 
-		print '
-
-	    <br>
-	    <form action="'.$_SERVER['REQUEST_URI'].'" method="POST" id="payment-form">';
+		//print '<br><form action="'.$_SERVER['REQUEST_URI'].'" method="POST" id="payment-form">';
 
 		print '<input type="hidden" name="token" value="'.$_SESSION['newtoken'].'">'."\n";
```

Comment:
This html code become orhpelin (without its form) if the form was previously removed.

Reply:
@eldy if we only moved form with stripe. is it ok for you ? payment intent is full JS and no need redirection for ok or ko message... customer can retry immediately or do 3Dsecure checking without refresh or redirection.

Reply:
@eldy i fix form for not stripe payment

Reply:
OK it is js. But if we remove form, when we click on the button "buttontopay", where is defined the page to go after ?
How stripe know if he has to call the page paymentok or paymentko ?

Reply:
with payment intent charge is done with JS (no more payment ko, if ko display directly an error message and customer can retry immediately) -> then payment in dolibarr is writed with webhook

we can also do manual integration but it's more complicated no webhook...  and need more code to support changes and 3D secure V2 ans strong authentication (be required in september 2019) -> downgrade to dolibarr v9 and V8 will be needed.

If we use the stripe JS, it manage it without further code (3Dsecure and/or biometric) in same page and  i will be more easy to add google/apple pay or iban form (sepa direct debit) in same form. 

https://stripe.com/docs/payments/payment-intents#one-time-payments

Reply:
So we must ask stripe to execute the webhooks and say him to execute the wekhook to then execute the paymentok.php (for exemple by making an 
include paymentok.php  if message if a charge success).

The url for webhook is defined into dolibarr setup. But i can't see where you provide the information to stripe. So how Stripe can call the webhook ?